### PR TITLE
Us040 login frontend

### PR DESCRIPTION
**Pull request by kelvinkainos**

### CODE REVIEW

File: .github/workflows/nightly-ui-tests-checks.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: "UI Tests"
+on:
+  push:
+    branches: [ "main" ]
```
Comment (glenn-hamilton) [inline .github/workflows/nightly-ui-tests-checks.yml]: These shouldn't be here, double check the `pull` and `pull_request` options are removed before merging

File: controller/rolesController.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { type Application, type Request, type Response } from 'express'
+
+const rolesService = require('../service/rolesService')
+
+module.exports = function (app: Application) {
```
Comment (glenn-hamilton) [inline controller/rolesController.ts]: Why is there `role` stuff in the MR?

File: public/styles.css [hunk 3]
```diff
@@ -54,10 +58,15 @@ body {
 footer {
     background-color: #283583;
     color: #fff;
+    position: fixed;
```
Comment (glenn-hamilton) [inline public/styles.css]: This has made the footer `sticky` is that intended?

File: test/ui/VIewRolesUITests.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/* eslint-env mocha */
```
Comment (glenn-hamilton) [inline test/ui/VIewRolesUITests.ts]: `VIewRolesUITests.ts` -> `ViewRolesUITests.ts`

File: test/ui/AuthLoginUITests.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            .getText()
+            .then(function (value) {
+                expect(value).equal(
+                    'Login details are invalid, please try again.'
+                )
+            })
+        await driver.findElement(By.id('email')).clear()
+        await driver.findElement(By.id('password')).clear()
+
+        // Test valid login
```
Comment (glenn-hamilton) [inline test/ui/AuthLoginUITests.ts:57]: Each of these scenarios could be separate tests

Comment (kelvinkainos) [reply test/ui/AuthLoginUITests.ts:57]: You said Selenium tests are too expensive, so I minimised them to one test case

File: views/index.html [hunk 6]
```diff
@@ -5,9 +5,9 @@
     <section>
         <div class="container-fluid hero">
             <div class="container-fluid smaller">
-                <button class="btn mt-2 hero-login" href="#">
-                    Sign in to view jobs
-                </button>
+                    <button class="btn mt-2 hero-login">
+                        <a href="/login" id="indexLogin">Log in to view jobs</a>
```
Comment (MorganMcDonald) [inline views/index.html:9]: Should this not be login not log in

**Review by MorganMcDonald — APPROVED** [review]

**Review by glenn-hamilton — DISMISSED** [review]

**Review by MorganMcDonald — DISMISSED** [review]

**Review by ThatOKPerson — APPROVED** [review]

**Review by ThatOKPerson — COMMENTED** [review]
Don't forget to squash commits

**Review by MorganMcDonald — APPROVED** [review]

**Pull request merged**